statsmodels/stats/multitest.py

```diff
@@ -444,3 +444,197 @@ def fdrcorrection_twostage(pvals, alpha=0.05, method='bky', iter=False,
         return reject, pvalscorr_, ntests - ri, alpha_stages
     else:
         return rej, pvalscorr, ntests - ri, alpha_stages
+
+
+def local_fdr(zscores, p0=1.0, null_density=None, deg=7,
+              nbins=30):
+    """
+    Calculate local FDR values for a list of Z-scores.
+
+    Arguments
+    ---------
+    zscores : array-like
+        A vector of Z-scores
+    p0 : float
+        The assumed proportion of true null hypotheses
+    null_density : function (real to positive real)
+        The density of null Z-scores; if None, use standard normal
+    deg : integer
+        The maximum exponent in the polynomial expansion of the
+        density of non-null Z-scores
+    nbins : integer
+        The number of bins for estimating the marginal density
+        of Z-scores.
+
+    Returns
+    -------
+    fdr : array-like
+        A vector of FDR values
+
+    References
+    ----------
+    B Efron (2008).  Microarrays, Empirical Bayes, and the Two-Groups
+    Model.  Statistical Science 23:1, 1-22.
+    """
+
+    from statsmodels.genmod.generalized_linear_model import GLM
+    from statsmodels.genmod.generalized_linear_model import families
+    from statsmodels.regression.linear_model import OLS
+
+    # Bins for Poisson modeling of the marginal Z-score density
+    minz = min(zscores)
+    maxz = max(zscores)
+    bins = np.linspace(minz, maxz, nbins)
+
+    # Bin counts
+    zhist = np.histogram(zscores, bins)[0]
+
+    # Bin centers
+    zbins = (bins[:-1] + bins[1:]) / 2
+
+    # The design matrix at bin centers
+    dmat = [(zbins**k)[:,None] for k in range(0, deg + 1)]
+    dmat = np.concatenate(dmat, axis=1)
+
+    # Use this to get starting values for Poisson regression
+    md = OLS(np.log(1 + zhist), dmat).fit()
+
+    # Poisson regression
+    md = GLM(zhist, dmat, family=families.Poisson()).fit(start_params=md.params)
+
+    # The design matrix for all Z-scores
+    dmat_full = [(zscores**k)[:,None] for k in range(0, deg + 1)]
+    dmat_full = np.concatenate(dmat_full, axis=1)
```

Comment:
same here np.vander
In general: if there is no special meaning to the units of the polynomial, then using normalized np.polynomial.vander or chebyvander would be numerically more stable. (might not matter here if zscores are standard normal and don't cover huge interval of numbers)